lab_3_generate_by_ngrams/main.py

```diff
@@ -309,6 +476,20 @@ def get_next_token(self, sequence: tuple[int, ...]) -> Optional[list[tuple[int,
 
         In case of corrupt input arguments or methods used return None.
         """
+        if not isinstance(sequence, tuple):
+            return None
+
+        tokens = self._model.generate_next_token(sequence)
+
+        if tokens is None:
+            return None
+
+        if not tokens:
+            return []
+
+        result = sorted([(token, float(freq)) for token, freq in tokens.items()],
+                        key=lambda x: x[1], reverse=True)
```

Comment:
why don't you cast to float in key?

Reply:
Because I suppose that tokens can't be float (and expected type is 'list[tuple[int, float]] | None')